sdk/storage/azure-storage-file/samples/file_samples_share_async.py

```diff
@@ -0,0 +1,128 @@
+                await share.create_snapshot()
+                # [END create_share_snapshot]
+            finally:
+                # [START delete_share]
+                await share.delete_share(delete_snapshots=True)
+                # [END delete_share]
+
+    async def set_share_quota_and_metadata_async(self):
+        # [START create_share_client_from_conn_string]
+        from azure.storage.file.aio import ShareClient
+        share = ShareClient.from_connection_string(self.connection_string, "sharesamples2")
+        # [END create_share_client_from_conn_string]
+
+        # Create the share
+        async with share:
+            await share.create_share()
+
+            try:
+                # [START set_share_quota]
+                # Set the quota for the share to 1GB
+                await share.set_share_quota(quota=1)
+                # [END set_share_quota]
+
+                # [START set_share_metadata]
+                data = {'category': 'test'}
+                await share.set_share_metadata(metadata=data)
+                # [END set_share_metadata]
+
+                # Get the metadata for the share
+                props = (await share.get_share_properties()).metadata
+
+            finally:
+                # Delete the share
+                await share.delete_share()
+
+    async def list_directories_and_files_async(self):
+        # Instantiate the ShareClient from a connection string
+        from azure.storage.file.aio import ShareClient
+        share = ShareClient.from_connection_string(self.connection_string, "sharesamples3")
+
+        # Create the share
+        async with share:
+            await share.create_share()
+
+            try:
+                # [START share_list_files_in_dir]
+                # Create a directory in the share
+                dir_client = await share.create_directory("mydir")
+
+                # Upload a file to the directory
+                with open(SOURCE_FILE, "rb") as source_file:
+                    await dir_client.upload_file(file_name="sample", data=source_file)
+
+                # List files in the directory
+                my_files = []
+                async for item in share.list_directories_and_files(directory_name="mydir"):
+                    my_files.append(item)
+                print(my_files)
+                # [END share_list_files_in_dir]
+            finally:
+                # Delete the share
+                await share.delete_share()
+
+    async def get_directory_or_file_client_async(self):
+        # Instantiate the ShareClient from a connection string
+        from azure.storage.file.aio import ShareClient
+        share = ShareClient.from_connection_string(self.connection_string, "sharesamples4")
+
+        # Get the directory client to interact with a specific directory
+        my_dir = share.get_directory_client("dir1")
+
+        # Get the file client to interact with a specific file
+        my_file = share.get_file_client("dir1/myfile")
```

Comment:
How does this behave when the directory may not exist?  If it handles it sanely/this is a normal use case then 100% ignore this comment, just being paranoid that we make any assumptions clear/that a user wouldn't be confused by "why doesn't this just let me do X operation on the file_client"